kanidmd/src/lib/idm/server.rs

```diff
@@ -1741,8 +1793,15 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
                     .contains(&UUID_SYSTEM_CONFIG)
                 {
                     self.reload_password_badlist(au)?;
-                    self.pw_badlist_cache.commit();
                 };
+                if self.qs_write.get_changed_ouath2() {
+                    self.qs_write
+                        .get_oauth2rs_set(au)
+                        .and_then(|oauth2rs_set| self.oauth2rs.reload(oauth2rs_set))?;
+                }
+                // Commit everything.
+                self.oauth2rs.commit();
+                self.pw_badlist_cache.commit();
```

Comment:
Wait can you give a quick rundown on what pieces of Oauth2 this file implements? I watched some videos on it but am still very confused.

Reply:
We are an identity provider. 

Consider an application like "gitter". When you go there is says "do you want to authenticate with github?". In this case the "resource server" is gitter, and kanidm would be "github", where we are providing identities and authorisation to external applications.

Reply:
So ideally in the future there would be options like "Sign in with Kanidm"?

Reply:
For resource servers, yes, they could add that option. More generically it's "sign in with openidconnect" or "sign in with oauth", which has more relevance inside an organisation that the public internet.